plugins/huddly-usb/fu-huddly-usb-device.c

```diff
@@ -400,10 +400,65 @@ fu_huddly_usb_device_hcp_write_file(FuHuddlyUsbDevice *self,
 	return fu_huddly_usb_device_hlink_unsubscribe(self, "hcp/write_reply", error);
 }
 
+static gboolean
+fu_huddly_usb_device_hpk_done_cb(FuDevice *device, gpointer user_data, GError **error)
+{
+	FuHuddlyUsbDevice *self = FU_HUDDLY_USB_DEVICE(device);
+	GPtrArray *items = (GPtrArray *)user_data;
+	GString *operation;
+	guint8 err;
+	gboolean need_reboot;
+	g_autoptr(FuHuddlyUsbHLinkMsg) msg_res = NULL;
+	g_autoptr(FuMsgpackItem) item_operation = NULL;
+	g_autoptr(FuMsgpackItem) item_error = NULL;
+	g_autoptr(FuMsgpackItem) item_reboot = NULL;
+
+	msg_res = fu_huddly_usb_device_hlink_receive(self, error);
+	if (msg_res == NULL)
+		return FALSE;
+	items = fu_msgpack_parse(msg_res->payload, error);
+	if (items == NULL)
+		return FALSE;
+	item_operation = fu_msgpack_map_lookup(items, 0, "operation", error);
+	if (item_operation == NULL)
+		return FALSE;
+	operation = fu_msgpack_item_get_string(item_operation);
+	g_debug("operation %s", operation->str);
+
+	/* get error */
+	item_error = fu_msgpack_map_lookup(items, 0, "error", error);
+	if (item_error == NULL)
+		return FALSE;
+	err = fu_msgpack_item_get_integer(item_error);
+	if (err != 0) {
+		g_prefix_error(error, "received error %s", operation->str);
+		return FALSE;
+	}
+
+	/* FIXME: do we need this? maybe store in private flag is so? */
```

Comment:
We always do a reboot when upgrading the software so it doesn't provide any useful information, but we may want to keep this and check it before rebooting just to be safe for the future.

Reply:
Can you elaborate a bit on why it's safer? Thanks.

Reply:
I don't know if the reboot flag is needed since the update always requires a reboot after writing the image. I will check with other team members what way is the safest